### ISSUE TITLE
Issues with `@include` / `@skip` on fragment spreads / inline fragments

### ISSUE THREAD

**Issue by BoD** [apollographql/apollo-kotlin#3810]
**Summary**
Fragment spreads / inline fragments with `@include(if: $variable)` are not parsed even when `$variable` is `true`.

**Version**
3.0.0

**Description**

When using `@include(if: true)` or `@include(if: false)` on a fragment spread or inline fragment, for instance:

```graphql
query LaunchList($cursor: String) {
  launches(after: $cursor) {
    cursor
    hasMore
    launches {
      ...launchFields @include(if: true)
    }
  }
}

fragment launchFields on Launch {
  id
  site
  mission {
    name
    missionPatch(size: SMALL)
  }
}
```

The generated code looks like:

```kotlin
// ...

reader.rewind()
var launchFields: LaunchFields? = null
if (BooleanExpression.True.evaluate(emptySet(), __typename)) {
  launchFields = com.example.rocketreserver.fragment.LaunchFieldsImpl_ResponseAdapter.LaunchFields.fromJson(reader,
      customScalarAdapters)
}

```

`BooleanExpression.True.evaluate(emptySet(), __typename)` evaluates to `true` so the fields of the `launchFields` fragment are correctly parsed (and inversely for `False`) ✅

However when using a variable:

```graphql
query LaunchList($cursor: String, $includeLaunchFields: Boolean!) {
  launches(after: $cursor) {
    cursor
    hasMore
    launches {
      ...launchFields @include(if: $includeLaunchFields)
    }
  }
}
```

The generated code looks like this:

```graphql
if (variable("includeLaunchFields").evaluate(emptySet(), __typename)) {
  launchFields = com.example.rocketreserver.fragment.LaunchFieldsImpl_ResponseAdapter.LaunchFields.fromJson(reader,
      customScalarAdapters)
}
```

`variable("includeLaunchFields").evaluate(emptySet(), __typename)` always return `false` (no matter the value of `$includeLaunchFields` ) so the fields are never parsed which is incorrect when `$includeLaunchFields` is `true`   ❌

### Possible fix

The generated adapter has no knowledge of the variables.

Therefore, it shouldn’t assume anything and always “try” to parse.

To do this, the code generated to evaluate a `BooleanExpression` must handle the `BVariable` case in such a way that calling `evaluate` on it returns true. One simple way to do this is to return  `BooleanExpression.True`.

However, making this change will now try to parse the fields even when they are not returned (the case when `$includeLaunchFields` is `false`). This will crash for any non nullable field in the fragment.

A possible (but not very satisfactory) fix is to add a `try catch` block and return `null`  if parsing fails.

In other words, the generated code could look like this:

```graphql
if (BooleanExpression.True.evaluate(emptySet(), __typename)) {
  launchFields = try { com.example.rocketreserver.fragment.LaunchFieldsImpl_ResponseAdapter.LaunchFields.fromJson(reader,
      customScalarAdapters) } catch (_: Exception) { null }
}
```

Of course having a `try catch` in the generated code is not very compelling... Could there be another solution I'm not seeing?
